src/electron/preload.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Directly import @sentry/electron main process code.
+// See: https://docs.sentry.io/platforms/javascript/guides/electron/#webpack-configuration
+
+import * as sentry from "@sentry/electron/dist/renderer";
+
+import {onceEnvVars} from "../www/app/environment";
+
+(async () => {
+  const envs = await onceEnvVars;
+  if (envs.SENTRY_DSN) {
```

Comment:
Why is this being done in the preload script, versus in the main?

Reply:
Actually we need to do both as noted in the [official doc](https://docs.sentry.io/platforms/javascript/guides/electron/#configure):

> You need to call init in the main process and every renderer process you spawn.

A `preload` script will be injected in every renderer process according to [Electron's document](https://www.electronjs.org/docs/latest/tutorial/process-model#preload-scripts).

If we don't do so (which means we only `init` in main), any sentry event from the renderer process (e.g., the user feedback web page) will be ignored.

Reply:
I'd like to avoid preload.ts whenever possible. That adds another component to build and makes things hard to reason about. In this case it's not needed.

We only have one renderer process. What I meant was for you to call that in the `main()` function of the renderer process. It seems we are already doing that: https://github.com/Jigsaw-Code/outline-client/blob/77ab9193092973e6bb3ca476e09d93bafa3b63b9/src/www/app/electron_main.ts#L122

Reply:
Yes, we already did the `init` in [`ElectronErrorReporter`](https://github.com/Jigsaw-Code/outline-client/blob/77ab9193092973e6bb3ca476e09d93bafa3b63b9/src/www/app/electron_main.ts#L79). But it seems this is not the renderer process because the following warning message will be printed out to the console (and we are not able to send user feedbacks as well):

> main.js:2 contextIsolation is enabled but IPC has not been exposed. Did you call "init" in the preload script?

Reply:
We call sentry.init already in the `main` process: https://github.com/Jigsaw-Code/outline-client/blob/77ab9193092973e6bb3ca476e09d93bafa3b63b9/src/electron/index.ts#L446
And in the `renderer` process: https://github.com/Jigsaw-Code/outline-client/blob/77ab9193092973e6bb3ca476e09d93bafa3b63b9/src/www/app/electron_main.ts#L79

I wonder if the `environment-info` event is triggering. Can you add debug info to make sure that's working?